inference-engine/src/mkldnn_plugin/mkldnn_graph_optimizer.cpp

```diff
@@ -319,14 +319,16 @@ void MKLDNNGraphOptimizer::FuseMultiplyAndAdd(MKLDNNGraph &graph) {
 
     auto isSutableParentNode = [&](MKLDNNNodePtr node) {
         if (node->getAlgorithm() != EltwiseMultiply || !node->getFusedWith().empty() ||
-            node->getParentEdges().size() != 2 || node->getChildEdges().size() != 1)
+            node->getParentEdges().size() != 2 || node->getChildEdges().size() != 1 ||
+            node->getParentEdgesAtPort(0)[0]->getDims().ndims() > MAX_ELTWISE_DIM_RANK)
             return false;
 
         return isSutableSecondInput(node->getParentEdgesAtPort(1)[0]->getParent(), node->getParentEdgesAtPort(0)[0]->getDims());
     };
 
     auto isSutableChildNode = [&](MKLDNNNodePtr parentNode, MKLDNNNodePtr childNode) {
-        if (childNode->getAlgorithm() != EltwiseAdd || !childNode->getFusedWith().empty() || childNode->getParentEdges().size() != 2)
+        if (childNode->getAlgorithm() != EltwiseAdd || !childNode->getFusedWith().empty() || childNode->getParentEdges().size() != 2 ||
+            childNode->getParentEdgesAtPort(0)[0]->getDims().ndims() > MAX_ELTWISE_DIM_RANK)
```

Comment:
Cannot we call EltwiseNode::canFuse to avoid this explicit condition?

Reply:
Yes, we can. Changed to `canFuse`, thanks!